### PR TITLE
Sdss 96 refactor integration tests

### PR DESCRIPTION
**Pull request by AND-JAshcroft**

### CODE REVIEW

File: src/app/config/config.py [hunk 1]
```diff
@@ -1,28 +1,62 @@
...
+        super().__init__()
+        self.CONF = get_value_from_env("CONF")
+        self.TIME_FORMAT = TIME_FORMAT_STRING
+        self.API_URL = get_value_from_env("API_URL", "localhost")
+        self.DATASET_BUCKET_NAME = get_value_from_env(
+            "DATASET_BUCKET_NAME", "testDatasetBucket"
+        )
+        self.SCHEMA_BUCKET_NAME = get_value_from_env(
+            "SCHEMA_BUCKET_NAME", "testSchemaBucket"
+        )
```
Comment (JackRimmerAND) [inline src/app/config/config.py:30]: Very minor but you're using camel case for these test vars but should really be snake case for Python test_schema_bucket

File: src/integration_tests/helpers/integration_helpers.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+    else:
+        bucket = storage.Client().bucket(config.DATASET_BUCKET_NAME)
+        blob = bucket.blob(filename)
+        blob.upload_from_string(
+            json.dumps(dataset, indent=2), content_type="application/json"
+        )
+
+        wait_until_dataset_ready(
+            dataset["survey_id"], dataset["period_id"], session, headers
+        )
```
Comment (JackRimmerAND) [inline src/integration_tests/helpers/integration_helpers.py:110]: Worth raising an error here if this doesn't work instead of returning 'None'?

File: src/integration_tests/helpers/integration_helpers.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+    Returns:
+        None
+    """
+    while attempts != 0:
+        test_response = session.get(
+            f"{config.API_URL}/v1/dataset_metadata?survey_id={survey_id}&period_id={period_id}",
+            headers=headers,
+        )
+        print(test_response.status_code)
```
[the remark this answers is not in the document]
Comment (AND-JAshcroft) [reply src/integration_tests/helpers/integration_helpers.py:139]: i did indeed

File: src/integration_tests/helpers/integration_helpers.py (hunk 2 above)
Comment (AND-JAshcroft) [reply src/integration_tests/helpers/integration_helpers.py]: Interestingly it already does as that method throws an exception if the dataset is not found

File: src/integration_tests/helpers/integration_helpers.py [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        if Path.is_dir(dataset_bucket_path):
+            shutil.rmtree(dataset_bucket_path)
+    else:
+        _delete_blobs(storage.Client().get_bucket(config.DATASET_BUCKET_NAME))
+
+        _delete_blobs(storage.Client().get_bucket(config.SCHEMA_BUCKET_NAME))
+
+        _delete_collection(db.collection("datasets"))
+
+        _delete_collection(db.collection("schemas"))
```
[the remark this answers is not in the document]
Comment (AND-JAshcroft) [reply src/integration_tests/helpers/integration_helpers.py:203]: As we chatted about earlier ive addresed the initial statement but kind of feel like these are fine the way they are

**Review by JackRimmerAND — APPROVED** [review]

File: cloudbuild.yaml [hunk 5]
```diff
@@ -51,8 +51,8 @@ steps:
         --source=. \
         --entry-point=new_dataset \
         --trigger-event-filters="type=google.cloud.storage.object.v1.finalized" \
-        --trigger-event-filters="bucket=${_DATASET_BUCKET}" \
-        --set-env-vars="SCHEMA_BUCKET_NAME=${_SCHEMA_BUCKET_NAME},CONF=cloud-build"
+        --trigger-event-filters="bucket=ons-sds-sandbox-01-europe-west2-dataset-892a" \
```
Comment (tidmanmike) [inline cloudbuild.yaml:54]: Why the change from variable to hardcoded value?

File: src/integration_tests/helpers/integration_helpers.py [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+    doc_ref.delete()
+
+
+MOCK_UNIT_RESPONSE = {
+    "schema_version": "v1.0.0",
+    "sds_schema_version": 4,
+    "survey_id": "test_survey_id",
+    "period_id": "abc",
+    "data": "<encrypted data>",
+}
```
Comment (IsaiahLai111) [inline src/integration_tests/helpers/integration_helpers.py:284]: Is it better if we put all the data, either the data that run through the test or to validate the test, into test data?

File: cloudbuild.yaml (hunk 5 above)
Comment (AND-JAshcroft) [reply cloudbuild.yaml:54]: ah thats a mistake thanks for spotting :)

File: src/integration_tests/datasets/e2e_dataset_integration_test.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    MOCK_UNIT_RESPONSE,
+    cleanup,
+    create_dataset,
+    generate_headers,
+    load_json,
+    setup_session,
+)
+
+config = ConfigFactory.get_config()
+UNIT_ID = "43532"
```
Comment (IsaiahLai111) [inline src/integration_tests/datasets/e2e_dataset_integration_test.py:16]: It might be better to put all the hard-coded data in testing in test data

File: src/integration_tests/helpers/integration_helpers.py [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+    Parameters:
+        filename: the filename to use for the dataset
+        dataset: the dataset to be created
+        session: a session instance for http/s connections
+        headers: the relevant headers for authentication for http/s calls
+
+    Returns:
+        int | None: status code for local function and no return for remote.
+    """
+    if config.API_URL.__contains__("local"):
```
Comment (IsaiahLai111) [inline src/integration_tests/helpers/integration_helpers.py:95]: I will suggest to refractor this logic in distinguishing a local test from remote test into a specific function within helpers

Comment (AND-JAshcroft) [reply src/integration_tests/helpers/integration_helpers.py:95]: Personally im not sure theres value in that, it would be a one line function that would make the flow of the helpers more complicated to follow

File: src/integration_tests/datasets/e2e_dataset_integration_test.py (hunk 7 above)
Comment (AND-JAshcroft) [reply src/integration_tests/datasets/e2e_dataset_integration_test.py:16]: Will move

File: src/integration_tests/helpers/integration_helpers.py (hunk 6 above)
Comment (AND-JAshcroft) [reply src/integration_tests/helpers/integration_helpers.py:284]: Will move

**Review by tidmanmike — APPROVED** [review]
